### ISSUE TITLE
gatherData not working properly

### ISSUE THREAD

**Issue by ermueller** [MassBank/RMassBank#101]
So, gatherData in createMassBank.R has some major issues as of right now. I noticed this when playing around with the glucolesquerellin data in the GUI. By extension, the mbWorkflow is broken for certain compounds.

I figure this issue is for Michele and me to solve.

I guess this happened because of CTS reworking everything?

The problems:
line 401: 
```R
    if(length(infos) == 0)
```

`infos` should technically be a list of all kinds of synonyms, but when CTS doesn't know anything about it, it doesn't return an empty list anymore, but rather the string `"Sorry, we couldn't find any matching results"`

so this should be: 
```R
     if(infos == "Sorry, we couldn't find any matching results")
```

Which is kind of strange, but it works, I guess.

The much bigger problem is from line 420 on:
if the name `Glucolesquerellin` is used, then this happens:
```R
> infos <- getCtsKey(dbname, from="Chemical Name", to="InChIKey")
> # heuristically determine best InChI key to use:
> # use the one with most common Structure part,
> # and use the one with no stereochemistry and neutral charge if possible
> keys <- as.data.frame(infos)
> subkeys <- strsplit(infos, ',')
> df <- do.call(rbind,subkeys)
> keys$structure <- df[,1]
> keys$stereo <- df[,2]
Fehler in df[, 2] : Indizierung außerhalb der Grenzen
> keys$charge <- df[,3]
Fehler in df[, 3] : Indizierung außerhalb der Grenzen
```

because the return structure of `infos` from "getCtsKeys" has been changed, I reckon. `infos` in this case just is a single data vector containing the InChIKey as a string, so no wonder it doesn't work. Splitting by commas does practically nothing, and from there on everything just stops working. I'm kind of at a loss on how to resolve this.

**Comment by ermueller** [comment]
Possible fix added in a comment in 1c3ba7b041e5e806140e0cd79759dd733eeb1aa5

**Comment by sneumann** [comment]
This fix can be enabled and thoroughly tested with the "christmas dataset"
